### ISSUE TITLE
Seafile.client seems to cause "Finder.app" to run on approx. 15% CPU usage constantly and to drain battery

### ISSUE THREAD

**Issue by GITchristoph** [haiwen/seafile-client#1042]
Hi

Macbook pro 13" with touchbar 2017 with High Sierra.

If Seafile client 6.1.3 is switched on, then the Finder.app requires 15% of CPU usage constantly and is marked as "App using significant energy".
If I quit Seafile client, then that behaviour stops.

It's fully reproducible on my Macbook.
Restart etc. did not fix it.

Is this only me or do other people experience the same?
Can I provide any log that could help to look into this issue?

Greetings, Chris

**Comment by GITchristoph** [comment]
Hi,

This is still the case for me on that same 2017 Macbook Pro 13, also with client 6.1.4.
I usually had to switch off Seafile client when on battery, which was a bit of a shame.

What seems to make it better though is to switch off the "Enable FinderSync Extension" setting.
That also helped on a 2014 Macbook Pro with MacOS Sierra. The impact on battery drainage was not so significant on that laptop though. Maybe this is based on the different Macbook models on not MacOS versions though.
In any case: Toggling this option on and off and watching the Activity Monitor on the Mac in the same time shows how the Finder requires more or less CPU usage respectively and instantly.

I hope this helps as a clue to solve the issue?
If there is anything I can provide from these laptops to help analyzing this and to figure out if this is a real issue, please let me know. 
I am glad to help.

Greetings, Chris

**Comment by mehlkopf** [comment]
This issue was also reported to the Seafile user forum (https://forum.seafile.com/t/mac-os-client-cpu-usage/2138/6), and it was buried under comments that suggested it was the user's system's fault, while the issue is 100 % reproducible on every Mac - I don't get it why the most severe performance bugs are not considered important enough to get fixed (at least on the Mac) …

**Comment by mehlkopf** [comment]
Although the general Seafile background CPU usage problem on the Mac was fixed with 6.2.3 (thank you very much for that, I really appreciate it!), this bug of the Finder extension is still present and renders the extension almost useless ...

**Comment by GITchristoph** [comment]
I must say that even on 6.2.3 with High Sierra, Seafile Client (without Finder extension) has the biggest energy impact of all my installed Apps. Especially when not connected to a Wifi (so no server connection), it's not clear why the Seafile Client continues to drain the battery.
Can this item not be improved?

**Comment by shoeper** [comment]
> Especially when not connected to a Wifi (so no server connection), 

Yes, I can also observe this on linux. There is some bug that most likely leads to fast infinite connection retries.

**Comment by shoeper** [comment]
There also is a dedicated issue https://github.com/haiwen/seafile-client/issues/1081
